src/cli/broker.toit

```diff
@@ -527,7 +534,8 @@ class Broker:
     upgrade-from := []
     if known-encoded-firmwares.is-empty:
       if base-firmwares.is-empty:
-        ui_.warning "Firmware of device '$device-id' is unknown. Upgrade might not use patches."
+        short := device-short-strings_[device-id]
```

Comment:
Maybe abstract getting the name of a device out, so you have one place to explain that device-short-strings_ must be set -- or check for null?

Reply:
done.